src/uart.rs

```diff
@@ -302,6 +303,116 @@ pub mod config {
         }
     }
 
+    /// Configures the interrupts the UART driver should enable
+    /// in order to be able to quickly inform us about the
+    /// related event.
+    #[derive(Debug, Clone)]
+    pub struct UserInterruptConfig {
+        /// If `Some(number_of_words)`, an interrupt will trigger
+        /// after `number_of_words` could have been transmitted
+        /// (unit is baudrate dependant).
+        ///
+        /// If `None` or `Some(0)` interrupt will be disabled.
+        pub receive_timeout: Option<u8>,
+        /// Sets the threshold **above** which an interrupt will
+        /// be generated (the hardware receive FIFO contains more words than
+        /// this number).
+        ///
+        /// If set to `None` interrupt will be disabled.
+        pub rx_fifo_full: Option<u8>,
+        /// Sets the threshold **below** which an interrupt will
+        /// be generated (the hardware transmit FIFO contains less words than
+        /// this number).
+        ///
+        /// If set to `None` interrupt will be disabled.
+        // TODO: what happens when set to 0?
+        pub tx_fifo_empty: Option<u8>,
+        /// Other interrupts to enable
+        pub flags: EnumSet<UserInterruptFlags>,
+        /// Allow using struct syntax,
+        /// but signal users other fields may be added
+        /// so `..Default::default()` should be used.
+        #[doc(hidden)]
+        pub _non_exhaustive: (),
+    }
+
+    impl Default for UserInterruptConfig {
+        fn default() -> Self {
```

Comment:
Nit: needs `pub const new` like everywhere and then the `Defalt` impl can delegate to `new`.